code/modules/urist/modules/GOAI/integrations/base_commander/mob_commander/obstacle_handlers/_helpers.dm

```diff
@@ -0,0 +1,75 @@
+//These are for actions that only really need to be defined once and used in many different handlers, but might need extra checks done inside a handler. These should all be called within a handler with an existing tracker, and passed any required parameters
+
+/**
+ * Checks if `pawn` is next to `target`, if not, starts navigation to it. Returns `TRUE` on completed navigation, `FALSE` if navigation is still in progress
+ */
+/datum/goai/mob_commander/proc/NavigateNextTo(var/datum/ActionTracker/tracker, var/atom/pawn, var/atom/target)
+	if(!tracker || !pawn || !target)
+		return
+
+	var/turf/target_turf = get_turf(target)
+
+	if(ChebyshevDistance(get_turf(pawn), target_turf) >= 2)
+		var/list/path_to_target = tracker.BBGet("PathToTarget", null)
+		if(isnull(path_to_target) || !src.active_path)
+			path_to_target = StartNavigateTo(target_turf, 1)
+			tracker.BBSet("PathToTarget", path_to_target)
+		return FALSE
+	else
+		return TRUE
+
+/**
+ * Checks if the tracker has timed-out, taking walking distance and tick delay into account. Takes an optional `additional_delay` arguement to allow for custom time-out durations
+ */
+/datum/goai/mob_commander/proc/TimedOutWalkDist(var/datum/ActionTracker/tracker, var/atom/pawn, var/atom/target, var/additional_delay = 0)
+	if(!tracker || !pawn || !target)
+		return
+
+	var/walk_dist = tracker.BBSetDefault("StartDist", (ManhattanDistance(get_turf(pawn), target) || 0)) || 0
+
+	if(tracker.IsOlderThan(src.ai_tick_delay * (10 + walk_dist + additional_delay)))
+		return TRUE
+	else
+		return FALSE
```

Comment:
You don't need an else branch here; just return FALSE by default and let successes overwrite the rval to TRUE.